build.go

```diff
@@ -180,11 +184,12 @@ func (c *Client) Build(ctx context.Context, opts BuildOptions) error {
 		return c.lifecycle.Execute(ctx, lifecycleOpts)
 	}
 
-	lifecycleImageSupported := lifecycleVersion.Equal(builder.VersionMustParse(prevLifecycleVersionSupportingImage)) || !lifecycleVersion.LessThan(semver.MustParse(minLifecycleVersionSupportingImage))
-
+	builderOS, err := lifecycleOpts.Builder.Image().OS()
```

Comment:
I'm confused... what's the point of getting `imgOS` and `builderOS` - could they be different?

Reply:
Ah, good catch. That was an oversight as the `imgOS` part was added at a later time. Will fix!